recipes/mpc/all/conanfile.py

```diff
@@ -16,65 +23,79 @@ class MpcConan(ConanFile):
-        self.requires("mpfr/4.1.0")
+        self.requires("gmp/6.2.1", transitive_headers=True)
+        self.requires("mpfr/4.1.0", transitive_headers=True)
 
     def validate(self):
-        # FIXME: add Visual Studio support, upstream has a makefile.vc
-        if self.settings.compiler == "Visual Studio":
-            raise ConanInvalidConfiguration("mpc can be built with Visual Studio, but it's not supported yet in this recipe.")
+        # FIXME: add msvc support, upstream has a makefile.vc
+        if is_msvc(self):
+            raise ConanInvalidConfiguration("mpc can be built with msvc, but it's not supported yet in this recipe.")
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
 
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        args = []
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append(f'--with-gmp={unix_path(self, self.dependencies["gmp"].package_folder)}')
+        tc.configure_args.append(f'--with-mpfr={unix_path(self, self.dependencies["mpfr"].package_folder)}')
         if self.options.shared:
-            args.extend(["--disable-static", "--enable-shared"])
+            tc.configure_args.extend(["--disable-static", "--enable-shared"])
         else:
-            args.extend(["--disable-shared", "--enable-static"])
-        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
+            tc.configure_args.extend(["--disable-shared", "--enable-static"])
+        tc.generate()
+
+        tc = AutotoolsDeps(self)
+        tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        autotools = self._configure_autotools()
+        apply_conandata_patches(self)
+        with chdir(self, self.source_folder):
+            if not os.path.exists("configure"):
+                command = "autoreconf -i"
+                self.run(command)
```

Comment:
The `Autotool` helper has an `.autoreconf()` method.
https://docs.conan.io/en/2.0/reference/tools/gnu/autotools.html#conan.tools.gnu.autotools.Autotools.autoreconf

Reply:
Updated as suggested